lib/galaxy/dependencies/dev-requirements.txt

```diff
@@ -1,99 +1,103 @@
+cmd2==1.5.0; python_version >= "3.6"
+colorama==0.4.4; python_version >= "3.6" and python_full_version < "3.0.0" and sys_platform == "win32" and python_version < "4.0" or sys_platform == "win32" and python_version >= "3.6" and python_full_version >= "3.5.0" and python_version < "4.0"
 coloredlogs==15.0; python_version >= "3.6" and python_full_version < "3.0.0" and python_version < "4" or python_version >= "3.6" and python_version < "4" and python_full_version >= "3.5.0"
-commonmark==0.9.1
-contextvars==2.4; python_version >= "3.6" and python_version < "3.7"
-coverage==5.3.1; python_version >= "2.7" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version < "4"
-cryptography==3.3.1; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version >= "3.6"
+commonmark==0.9.1; python_version >= "3.6" and python_version < "4.0"
+contextvars==2.4; python_version < "3.7" and python_version >= "3.6"
+coverage==5.5; python_version >= "2.7" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version < "4"
+cryptography==3.4.6; python_version >= "3.6"
 cwltool==3.0.20201109103151; python_version >= "3.6" and python_version < "4"
 dataclasses==0.8; python_version >= "3.6" and python_version < "3.7"
 debtcollector==2.2.0; python_version >= "3.6"
 decorator==4.4.2; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.2.0" and python_version >= "3.6"
-defusedxml==0.7.0rc2; python_version >= "3.0" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version >= "3.0"
-deprecated==1.2.11; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.6"
+defusedxml==0.7.1; python_version >= "3.0" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version >= "3.0"
+deprecated==1.2.11; python_version >= "3.5" and python_full_version < "3.0.0" or python_full_version >= "3.4.0" and python_version >= "3.5"
 deprecation==2.1.0
 dictobj==0.4
 docopt==0.6.2
 docutils==0.16; (python_version >= "2.7" and python_full_version < "3.0.0") or (python_full_version >= "3.5.0")
-dogpile.cache==1.1.1; python_version >= "3.6"
+dogpile.cache==1.1.2; python_version >= "3.6"
 ecdsa==0.14.1; python_version >= "2.6" and python_full_version < "3.0.0" or python_full_version >= "3.3.0"
 fabric3==1.14.post1
 fastapi-utils==0.2.1; python_version >= "3.6" and python_version < "4.0"
-fastapi==0.63.0; python_version >= "3.6"
-fluent-logger==0.10.0
+fastapi==0.1.17; python_version >= "3.6"
```

Comment:
This downgrade doesn't look good at all

Reply:
No. That's not good at all. Same in the newly generated lock file. Also, there's the numpy error (I've seen it recently after running the same update). https://github.com/galaxyproject/galaxy/runs/2102114198#step:6:1248
Do you think we should consider adding reasonable minimal version constraints to all dependencies in the pyproject.toml file?

Reply:
> Do you think we should consider adding reasonable minimal version constraints to all dependencies in the pyproject.toml file?

If there is a real requirements (e.g. something in Galaxy won't work with an older version), a `>=` version constraint is absolutely fine in my opinion (and may reduce the search space and the time spent by poetry).

Reply:
Oh yes, of course, I agree - for individual cases where we know of a specific version boundary below which we can't go. But setting arbitrary minimum constrains for all our 91 dependencies would, most likely, overconstrain.

Reply:
In this particular case, we could find a reasonable minimum FastAPI minimum version to steer the solver.

Reply:
> Also, there's the numpy error (I've seen it recently after running the same update). 

This should be fixed by [bx-python 0.8.11](https://pypi.org/project/bx-python/0.8.11/), which now uses [oldest-supported-numpy](https://pypi.org/project/oldest-supported-numpy/) to build its wheels: https://github.com/bxlab/bx-python/commit/22d9a2506aeca7489ec89f51a1d6f552960f3fb2

Reply:
Trying this locally now. Starting with ">=0.63.0": this was release on December 20, which was before most of our changes were made - so that's the version we've been actively using. (I wonder if this might be a reasonable starting point for similar cases - to set the minimal allowed version to the version that was used in development, or as close to it as possible)